### PR TITLE
feat: support new MLS cipher suite [WPB-8592]

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

1. a new MLS cipher suite need to be supported
2. the info about the cipher suite if customizable by the backend

### Solutions

parse and store the default cipher suite info from the server  and use it to start the MLS client

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/CoreCryptoCentral.kt [hunk 1]
```diff
@@ -61,12 +71,17 @@ private class Callbacks : CoreCryptoCallbacks {
...
+    private val cipherSuite: Ciphersuites,
+    private val defaultCipherSuite: UShort?
+) : CoreCryptoCentral {
     fun getCoreCrypto() = cc
 
     override suspend fun mlsClient(clientId: CryptoQualifiedClientId): MLSClient {
-        cc.mlsInit(clientId.toString().encodeToByteArray(), Ciphersuites.DEFAULT.lower(), null)
-        return MLSClientImpl(cc)
+        cc.mlsInit(clientId.toString().encodeToByteArray(), cipherSuite, null)
+        return MLSClientImpl(cc, defaultCipherSuite!!)
```
Comment (mchenani) [inline cryptography/src/commonJvmAndroid/kotlin/com.wire.kalium.cryptography/CoreCryptoCentral.kt:84]: suggestion: add a todo to remove it later when CC removed it from it's API?

File: cryptography/src/commonTest/kotlin/com/wire/kalium/cryptography/E2EIClientTest.kt [hunk 2]
```diff
@@ -35,7 +35,7 @@ class E2EIClientTest : BaseMLSClientTest() {
     }
 
     private suspend fun createE2EIClient(user: SampleUser): E2EIClient {
-        return createMLSClient(user.qualifiedClientId).e2eiNewActivationEnrollment(
+        return createMLSClient(user.qualifiedClientId, listOf(1.toUShort()), 1.toUShort()).e2eiNewActivationEnrollment(
```
Comment (mchenani) [inline cryptography/src/commonTest/kotlin/com/wire/kalium/cryptography/E2EIClientTest.kt:38]: suggestion: define these values then we can change them later easier?

File: cryptography/src/commonTest/kotlin/com/wire/kalium/cryptography/MLSClientTest.kt [hunk 3]
```diff
@@ -33,7 +33,7 @@ class MLSClientTest : BaseMLSClientTest() {
     }
 
     private suspend fun createClient(user: SampleUser): MLSClient {
-        return createMLSClient(user.qualifiedClientId)
+        return createMLSClient(user.qualifiedClientId, listOf(1.toUShort()), 1.toUShort())
```
Comment (mchenani) [inline cryptography/src/commonTest/kotlin/com/wire/kalium/cryptography/MLSClientTest.kt:36]: suggestion: here the same, define some variables.

File: cryptography/src/commonTest/kotlin/com/wire/kalium/cryptography/E2EIClientTest.kt (hunk 2 above)
Comment (MohamadJaara) [reply cryptography/src/commonTest/kotlin/com/wire/kalium/cryptography/E2EIClientTest.kt:38]: i was thinking about a way to test all possible ciphersuits

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/mlsmigration/MLSMigrationWorker.kt [hunk 4]
```diff
@@ -60,7 +60,7 @@ internal class MLSMigrationWorkerImpl(
 
     private suspend fun syncMigrationConfigurations(): Either<CoreFailure, Unit> =
         featureConfigRepository.getFeatureConfigs().flatMap { configurations ->
-            mlsConfigHandler.handle(configurations.mlsModel, duringSlowSync = false)
+            mlsConfigHandler.handle(TODO(), duringSlowSync = false)
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/mlsmigration/MLSMigrationWorker.kt:63]: missing something here?

**Review by mchenani — APPROVED** [review]
LGTM, added some suggestion and spotted one forgotten TODO?

**Review by typfel — CHANGES_REQUESTED** [review]
Note you need to:
- Provide the matching external sender for the cipher-suite when creating new MLS group
- ~~Upload the matching public key with the right name when registering an MLS client~~

Maybe I missed it but I don't see that being done in this PR.

**Comment by MohamadJaara** [comment]
> Upload the matching public key with the right name when registering an MLS client

this is done here

**Comment by typfel** [comment]
> > Upload the matching public key with the right name when registering an MLS client
> 
> this is done here #2720

Cool, what about the external senders when creating a group?

**Comment by MohamadJaara** [comment]
> > > Upload the matching public key with the right name when registering an MLS client
> > 
> > 
> > this is done here #2720
> 
> Cool, what about the external senders when creating a group?

t is already included
https://github.com/wireapp/kalium/pull/2716/files#diff-7ff2f388e622ca17f7b1a9be76851c4ae48fe928ccce10419932b3536eabcb56R110

**Pull request merged**